### ISSUE TITLE
use cmake?

### ISSUE THREAD

**Issue by beOn** [fpbattaglia/PythonPlugin#7]
I've got a decent cmake file for building plugins... see [my fork of ZMQInterface](https://github.com/beOn/ZMQInterface). Nevermind the rest of what I've done to that plugin...

Need to put some attribution at the top of the file, but other than that I'd be happy to adapt it for this plugin. You use ccmake to tell it where open ephys is, and it selects the correct install location for the platform (on macs, this is under ~/Application\ Support/open-ephys/PlugIns, not in the open ephys build path).

It can make xcode projects, make files, and project files for whatever IDE's cmake supports (though I've only tested those first two).

It can also generate customized environment.xcconfig files, so should take care of that pesky python path / version issue in the xcode build process.

Let me know if you'd like me to put something together!

**Comment by Claybarn** [comment]
Definitely! The python path issue is a real pain. Would be awesome to get around that. Be sure to develop with the “event_reciever” branch.

**Comment by ethanbb** [comment]
To add on, there is now this standard template for using CMake for Open Ephys plugins: https://github.com/open-ephys-plugins/OEPlugin

I haven't looked at @beOn's fork, but it sounds like it might be possible to incorporate the logic for setting the Python path into a CMake config file based on the template, and use that to replace the various python path setting mechanisms for each platform (e.g. `build-linux.sh` for Linux and `PythonEnv.props` for Windows).

**Comment by fpbattaglia** [comment]
Always tried to avoid cmake, but if somebody understands it well enough, and it can deal with the paths stuff etc, then that's great

**Comment by beOn** [comment]
I've got it up and running on the mac build, and it's pretty super.

Working on the linux build... and I could use a volunteer to help testing
out the windows build process.

**Comment by fpbattaglia** [comment]
just created a "cmake_build" branch. Do you want to send a PR onto that? Then we can all take a look...

**Comment by beOn** [comment]
Sure thing. Will have to hit this tonight or over the weekend... updating my branch from master just now created a bit of a mess.

**Comment by beOn** [comment]
Submitted. Some quick notes I'll need to edit, extend and add to the readme:

readme notes
============
- need to install cmake (link for platform-specific instructions)
- cmake build should be run using whatever python environment you're gonna use
    - or specify PYTHON_EXECUTABLE, PYTHON_LIBRARIES, PYTHON_INCLUDE_DIRS, PYTHON_HOME
    - example cmake command: cmake ../ -G Xcode -DOPEN_EPHYS_DIR=../../openEphysGUI
- they seem to use the root anaconda version... we'll fix this
- note that if you're on a mac, and you've built the plugin previously, you should delete the stuff it put in your open ephys dir
- after cmake, you need to build:
    + xcode
        * xcodebuild -target install build >/dev/null
        * or open the project and build it
